File: MovieRaterApi/Data/Entities/Rating.cs
namespace MovieRaterApi.Data.Entities;

public class Rating
{
    public Guid Id { get; set; }
    public Guid WatchSessionId { get; set; }
    public Guid UserId { get; set; }
    public int RatingValue { get; set; }
    public string? Review { get; set; }
    public DateTime CreatedAt { get; set; }
    public DateTime UpdatedAt { get; set; }

    public WatchSession WatchSession { get; set; } = null!;
    public User User { get; set; } = null!;
}

File: movie-rater-fe/src/types/rating.ts
export interface RatingResponseDto {
  id: string
  userId: string
  username: string
  ratingValue: number
  review: string | null
  createdAt: string
  updatedAt: string
}

export interface SessionRatingsResponseDto {
  watchSessionId: string
  ratings: RatingResponseDto[]
}

export interface CreateRatingRequestDto {
  ratingValue: number
  review?: string | null
}

export interface UpdateRatingRequestDto {
  ratingValue: number
  review?: string | null
}